lib/datadog/statsd.rb

```diff
@@ -425,14 +425,27 @@ def flush_buffer()
       @buffer = Array.new
     end
 
+    def connect_to_socket(host, port)
+      socket = UDPSocket.new
+      socket.connect(host, port)
+      socket
+    rescue => boom
```

Comment:
My rationale for that `rescue` was to make a transparent change which wouldn't affect any of current behavior, so this could be shipped as a patch version update. 
Currently there is no way that `Datadog::Statsd.new` would raise an exception. While if I would omit this rescue this won't be true any more. So I don't mind to remove this rescue since this PR is aimed for next minor version upgrade and such change is acceptable.